ch10.md

```diff
@@ -325,97 +317,21 @@ A.of(compose).ap(u).ap(v).ap(w) == u.ap(v.ap(w));
 ```
 
 ```js
-var u = IO.of(_.toUpper);
-var v = IO.of(_.concat("& beyond"));
-var w = IO.of("blood bath ");
+const u = IO.of(toUpper);
+const v = IO.of(concat("& beyond"));
+const w = IO.of("blood bath ");
 
-IO.of(_.compose).ap(u).ap(v).ap(w) == u.ap(v.ap(w))
+IO.of(compose).ap(u).ap(v).ap(w) == u.ap(v.ap(w))
 ```
 
 ## In Summary
 
 A good use case for applicatives is when one has multiple functor arguments. They give us the ability to apply functions to arguments all within the functor world. Though we could already do so with monads, we should prefer applicative functors when we aren't in need of monadic specific functionality.
 
-We're almost finished with container APIs. We've learned how to `map`, `chain`, and now `ap` functions. In the next chapter, we'll learn how to work better with multiple functors and disassemble them in a principled way.
+We're almost finished with container apis. We've learned how to `map`, `chain`, and now `ap` functions. In the next chapter, we'll learn how to work better with multiple functors and disassemble them in a principled way.
 
 [Chapter 11: Traversable/Foldable Functors](ch11.md)
 
 
-
 ## Exercises
-
-```js
-require('./support');
-var Task = require('data.task');
-var _ = require('ramda');
-
-// fib browser for test
-var localStorage = {};
-
-
-
-// Exercise 1
-// ==========
-// Write a function that adds two possibly null numbers together using Maybe and ap().
-
-//  ex1 :: Number -> Number -> Maybe Number
-var ex1 = function(x, y) {
-  // write me
-};
-
-
-// Exercise 2
-// ==========
-// Now write a function that takes 2 Maybe's and adds them. Use liftA2 instead of ap().
-
-//  ex2 :: Maybe Number -> Maybe Number -> Maybe Number
-var ex2 = undefined;
-
-
-
-// Exercise 3
-// ==========
-// Run both getPost(n) and getComments(n) then render the page with both. (The n arg is arbitrary.)
-var makeComments = _.reduce(function(acc, c) { return acc+"<li>"+c+"</li>" }, "");
-var render = _.curry(function(p, cs) { return "<div>"+p.title+"</div>"+makeComments(cs); });
-
-//  ex3 :: Task Error HTML
-var ex3 = undefined;
-
-
-
-// Exercise 4
-// ==========
-// Write an IO that gets both player1 and player2 from the cache and starts the game.
-localStorage.player1 = "toby";
-localStorage.player2 = "sally";
-
-var getCache = function(x) {
-  return new IO(function() { return localStorage[x]; });
-};
-var game = _.curry(function(p1, p2) { return p1 + ' vs ' + p2; });
-
-//  ex4 :: IO String
-var ex4 = undefined;
-
-
-
-
-
-// TEST HELPERS
-// =====================
-
-function getPost(i) {
-  return new Task(function (rej, res) {
-    setTimeout(function () { res({ id: i, title: 'Love them futures' }); }, 300);
-  });
-}
-
-function getComments(i) {
-  return new Task(function (rej, res) {
-    setTimeout(function () {
-      res(["This book should be illegal", "Monads are like space burritos"]);
-    }, 300);
-  });
-}
-```
+[include](./code/applicative/exercises.js)
```

Comment:
Checkout the project.
```bash
npm install
npm run setup
npm start
```
Read the book in the browser.
Basically I want to move to a place where we can have runnable and testable code and the splice it into the book.